### PR TITLE
PCVL-759-kl divergence calc

### PR DESCRIPTION
**Pull request by raksharuia**

### CODE REVIEW

File: tests/test_dist_metrics.py [hunk 1]
```diff
@@ -64,3 +64,13 @@ def test_tvd_one_empty_dist():
     target_bsd[bs2] = 0.7
 
     assert tvd_dist(target_bsd, BSDistribution()) == 0.5
+
+
+def test_kl_div_identical_dist():
```
Comment (Aubaert) [inline tests/test_dist_metrics.py:69]: Add a test with non-identical dist ?

File: perceval/utils/dist_metrics.py [hunk 2]
```diff
@@ -50,3 +50,32 @@ def tvd_dist(dist_lh: BSDistribution, dist_rh: BSDistribution) -> float:
...
+
+    :param ideal_dist: Ideal BSDistribution (known from theory or an ideal computation)
+    :param est_dist: Estimated BSDistribution (simulated or observed from experiment)
+    :return : KL divergence of the estimated distribution relative to the ideal.
+    """
+
+    kl_div = 0
+    zero_states_count = 0  # states with null probabilities or missing in estimated distribution
+
+    for states, ideal_probs in ideal_dist.items():
```
Comment (ericbrts) [inline perceval/utils/dist_metrics.py:69]: Variables you retrieve by iterating on the distribution should be singular : `state, ideal_prob`

Below `est_prob`

File: tests/test_dist_metrics.py [hunk 3]
```diff
@@ -64,3 +65,25 @@ def test_tvd_one_empty_dist():
...
+def test_kl_div_unequal_dist():
+    ideal_bsd = BSDistribution()
+    ideal_bsd[bs1] = 0.25
+    ideal_bsd[bs2] = 0.5
+
+    model_bsd = BSDistribution()
+    model_bsd[bs1] = 1
+    model_bsd[bs2] = 0.7
+
+    assert kl_divergence(ideal_bsd, model_bsd) != 0
```
Comment (ericbrts) [inline tests/test_dist_metrics.py:89]: Isn't != a bit too broad? Isn't there an easy expected value between two distributions that we could use to check the result value?

**Review by ericbrts — APPROVED** [review]

File: perceval/utils/dist_metrics.py [hunk 4]
```diff
@@ -50,3 +50,32 @@ def tvd_dist(dist_lh: BSDistribution, dist_rh: BSDistribution) -> float:
...
+
+def kl_divergence(ideal_dist: BSDistribution, est_dist: BSDistribution) -> float:
+    """
+    Computes the Kullback-Leibler (KL) divergence of a model (simulated/observed) BSdistribution with respect to
+     an ideal BSDistribution.
+     Our computation ignores states absent from the estimated distribution or have null probabilities.
+
+    :param ideal_dist: Ideal BSDistribution (known from theory or an ideal computation)
+    :param est_dist: Estimated BSDistribution (simulated or observed from experiment)
+    :return : KL divergence of the estimated distribution relative to the ideal.
```
Comment (ericbrts) [inline perceval/utils/dist_metrics.py:63]: remove the space in `:return :`

**Pull request merged**